lib/ibmdb.js

```diff
@@ -539,113 +539,203 @@ IBMDB.prototype.updateOrCreate = IBMDB.prototype.save =
  * @param {Object} data The model instance data
  * @param {Function} [callback] The callback function
  */
-IBMDB.prototype.replaceOrCreate = IBMDB.prototype.replace =
-  function(model, data, options, callback) {
-    debug('IBMDB.prototype.replaceOrCreate (enter): model=%j, data=%j, ' +
-          'options=%j\n', model, data, options);
-    var self = this;
-    var idName = self.idName(model);
-    var stmt;
-    var tableName = self.tableEscaped(model);
-    var meta = {};
-
-    function executeWithConnection(connection, cb) {
-      // Execution for replaceOrCreate requires running 3
-      // separate SQL statements. The last depends on the
-      // result of the first couple.
-      var where = {};
-      where[idName] = data[idName];
-
-      var countStmt = new ParameterizedSQL('SELECT COUNT(*) AS CNT FROM ');
-      countStmt.merge(tableName);
-      countStmt.merge(self.buildWhere(model, where));
-      countStmt.noResults = false;
-
-      connection.query(countStmt, function(err, countData) {
-        debug('IBMDB.prototype.replaceOrCreate (data): err=%j, countData=%j\n',
-              err, countData);
-        if (err) return cb(err);
+IBMDB.prototype._replace = function(model, where, data, options, callback) {
+  debug('IBMDB.prototype._replace (enter): model=%j, data=%j, ' +
+        'options=%j\n', model, data, options);
+  var self = this;
+  var idName = self.idName(model);
```

Comment:
Fixed.  Caught once again by lint running after tests and not thinking to run it separately.